subprojects/plugins/src/main/java/org/gradle/jvm/component/SingleTargetJvmFeature.java

```diff
@@ -72,12 +60,12 @@ public interface JvmFeatureInternal {
      */
     void withSourcesJar();
 
+    // TODO: Should this live on the "base" JVM feature? Should all JVM features know how to add
```

Comment:
I think it should move to a subclass for exactly these reasons.

Reply:
Agreed. Only Java features should have an API for javadoc. Though, as it stands in Gradle all features are currently Java features. Even groovy and scala projects have a compileJava and javadoc tasks. I've reworded the TODO to state we should move this to a java-specific model, so that it is no longer a question.